Content.Server/Dice/DiceSystem.cs

```diff
@@ -1,81 +1,26 @@
-using Robust.Shared.Audio;
-using Robust.Shared.Player;
 using Robust.Shared.Random;
 
-namespace Content.Server.Dice
-{
-    [UsedImplicitly]
-    public sealed class DiceSystem : EntitySystem
-    {
-        [Dependency] private readonly IRobustRandom _random = default!;
-
-        public override void Initialize()
-        {
-            base.Initialize();
-
-            SubscribeLocalEvent<DiceComponent, ComponentInit>(OnComponentInit);
-            SubscribeLocalEvent<DiceComponent, UseInHandEvent>(OnUseInHand);
-            SubscribeLocalEvent<DiceComponent, LandEvent>(OnLand);
-            SubscribeLocalEvent<DiceComponent, ExaminedEvent>(OnExamined);
-        }
-
-        private void OnComponentInit(EntityUid uid, DiceComponent component, ComponentInit args)
-        {
-            if (component.CurrentSide > component.Sides)
-                component.CurrentSide = component.Sides;
-        }
-
-        private void OnUseInHand(EntityUid uid, DiceComponent component, UseInHandEvent args)
-        {
-            if (args.Handled) return;
-
-            args.Handled = true;
-            Roll(uid, component);
-        }
+namespace Content.Server.Dice;
 
-        private void OnLand(EntityUid uid, DiceComponent component, LandEvent args)
-        {
-            Roll(uid, component);
-        }
-
-        private void OnExamined(EntityUid uid, DiceComponent dice, ExaminedEvent args)
-        {
-            //No details check, since the sprite updates to show the side.
-            args.PushMarkup(Loc.GetString("dice-component-on-examine-message-part-1", ("sidesAmount", dice.Sides)));
-            args.PushMarkup(Loc.GetString("dice-component-on-examine-message-part-2", ("currentSide", dice.CurrentSide)));
-        }
-
-        public void SetCurrentSide(EntityUid uid, int side, DiceComponent? die = null, SpriteComponent? sprite = null)
-        {
-            if (!Resolve(uid, ref die, ref sprite))
-                return;
-
-            side = Math.Min(Math.Max(side, 1), die.Sides);
-            side += side % die.Step;
-
-            die.CurrentSide = side;
-
-            // TODO DICE: Use a visualizer instead.
-            sprite.LayerSetState(0, $"d{die.Sides}{die.CurrentSide}");
-        }
+[UsedImplicitly]
+public sealed class DiceSystem : SharedDiceSystem
+{
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly SharedPopupSystem _popup = default!;
 
-        public void Roll(EntityUid uid, DiceComponent? die = null)
-        {
-            if (!Resolve(uid, ref die))
-                return;
+    public override void Roll(EntityUid uid, DiceComponent? die = null)
+    {
+        if (!Resolve(uid, ref die))
+            return;
 
-            var roll = _random.Next(1, die.Sides/die.Step+1) * die.Step;
-            SetCurrentSide(uid, roll, die);
+        var roll = _random.Next(1, die.Sides);
```

Comment:
Old code was correct here. This won't work for d100 since it doesn't actually have 100 sides.

Reply:
I see now in the PR description that you want it to work like an actual 100-sided dice when examined, but not visually? Can you explain why you think the number on the sprite shouldn't match the number shown when examined for this particular dice?

Reply:
> Can you explain why you think the number on the sprite shouldn't match the number shown when examined for this particular dice?

They should. The main point of this PR is really the last point: getting rid of the server sprite references.

Making the d100 not just a d10 in disguise & making the numbers legible was just a tangent. IMO it should actually behave as a proper d100 including the sprite, but that can be done by some other PR because that required a not insignificant amount of work.

Reply:
If you really want an "actual d100", why can't we keep the old one and have both, so that those that prefer the more standard d100 can use the old one?

Reply:
I'm not sure why you'd want to keep the old one? There already is a d10 in game and having a duplicate d100 that can't actually function as a d100 just seems unnecessarily confusing.

Reply:
I don't understand what you mean by "can't actually function as a d100". It functions just fine as it is now, and your change just makes the examine text not match the result on the sprite, which seems a bit confusing to me. Can you explain what actual problem are you trying to fix by changing the way d100s work?

Reply:
I think there is some confusion for terminology. d100 to me means a literal 100-sided die, but apparently some people use d100 to refer to percentile dies where you roll it in combination with a d10. In hindsight & after reading the entity description I guess this is what the die was meant to be so I'll revert to the old behaviour and just rename it to avoid confusion with any actuall d100s that might get added in future.

Reply:
I also changed the pop-up to show "0" instead of "100" when the dice shows "00". Or is "100" just the standard notation?

Also not sure whether "percentile" or "10s die" or something else is better. I'm just opposed to calling it a d100

Reply:
Showing 0 instead of 100 seems reasonable since it's only a 100 if you also roll 0 on the d10.

I don't have a strong opinion on the name. Both d100 and percentile are common names for it.